pkg/client/delegation/redelegation.go

```
@@ -50,37 +55,38 @@ func newRedelegationEventFactoryFn() events.NewEventsFn[client.Redelegation] {
 			return nil, err
 		}
 		// Check if the TxEvent has empty transaction bytes, which indicates
-		// the message might not be a valid transaction event.
-		if bytes.Equal(txEvent.Tx, []byte{}) {
-			return nil, events.ErrEventsUnmarshalEvent.Wrapf("%s", string(eventBz))
+		// the message is probably not a valid transaction event.
+		if len(txEvent.Tx) == 0 {
+			return nil, events.ErrEventsUnmarshalEvent.Wrap("empty transaction bytes")
 		}
 		// Iterate through the log entries to find EventRedelegation
 		for _, event := range txEvent.Result.Events {
-			if event.GetType_() == "pocket.application.EventRedelegation" {
-				var redelegationEvent redelegation
-				for _, attr := range event.Attributes {
-					switch attr.Key {
-					case "app_address":
-						appAddr, err := unescape(attr.Value)
-						if err != nil {
-							return nil, events.ErrEventsUnmarshalEvent.Wrapf("%s", string(eventBz))
-						}
-						redelegationEvent.AppAddress = appAddr
-					case "gateway_address":
-						gatewayAddr, err := unescape(attr.Value)
-						if err != nil {
-							return nil, events.ErrEventsUnmarshalEvent.Wrapf("%s", string(eventBz))
-						}
-						redelegationEvent.GatewayAddress = gatewayAddr
+			if event.GetType_() != redelegationEventType {
+				continue
+			}
+			var redelegationEvent redelegation
+			for _, attr := range event.Attributes {
+				switch attr.Key {
+				case "app_address":
+					appAddr, err := unescape(attr.Value)
+					if err != nil {
+						return nil, events.ErrEventsUnmarshalEvent.Wrapf("cannot retrieve app address: %v", err)
 					}
+					redelegationEvent.AppAddress = appAddr
+				case "gateway_address":
```

Can you add a default that logs an error or panics so it's not a silent error?